### PR TITLE
fix(core): fix vue 2 issues, closes #799

### LINKED ISSUE #799: [Vuelidate 2] [Vue warn]: Error in data(): "TypeError: Cannot read property 'validations' of undefined"
### Make sure that you are familiar with documentation before submitting an issue.
following the snippet from https://vuelidate-next.netlify.app/#getting-started-2, using `@vue/composition-api 1.0.0-rc.1`
  https://vuelidate.netlify.com/

### Please use the following jsfiddle to provide a reproduction of your problem

  https://codepen.io/metalsadman/pen/MWbjvJm?editors=1010

Steps:
 1. open provided codepen
 2. un-comment line `// v$: useVuelidate()`
 3. see error logs

### PR DESCRIPTION
**Pull request by dobromir-hristov**
## Summary

Enables testing Vue 2 and Vue 3 at the same time. Fixes issues with Vue 2. Closes #799 

## Breaking Changes

None

### CODE REVIEW

File: packages/vuelidate/src/core.js [hunk 1]
```diff
@@ -501,7 +501,7 @@ export function setValidations ({
       const s = unwrap(state)
       return s ? unwrap(s[key]) : undefined
     })
-    : isProxy(state) ? state : reactive(state || {})
+    : state
```
Comment (dobromir-hristov) [inline packages/vuelidate/src/core.js:504]: This was useful if you want to watch the top level state, which I think we dont really need. 99% of people wont replace the entire state.

File: packages/vuelidate/test/unit/test-utils.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const { isVue3 } = require('vue-demi')
+
+module.exports = isVue3 ? require('@vue/test-utils') : require('@vue/test-utils-vue2')
```
Comment (dobromir-hristov) [inline packages/vuelidate/test/unit/test-utils.js:3]: cool, ha ? :D

Comment (shentao) [reply packages/vuelidate/test/unit/test-utils.js:3]: Impressive 😯

**Review by shentao — APPROVED** [review]
Looks good! Nice work!

**Pull request merged**